lib/consul/controller.rb

```diff
@@ -105,18 +68,26 @@ def power(*args)
 
       end
 
+      # On first access we inherit .consul_power_args from our ancestor classes.
+      # We also copy inherited args so we don't change our parent's .consul_power_args
+      def consul_power_args
+        unless @consul_power_args_initialized
+          if superclass && superclass.respond_to?(:consul_power_args, true)
+            @consul_power_args = superclass.send(:consul_power_args).dup
+          else
+            @consul_power_args = []
+          end
+          @consul_power_args_initialized = true
+        end
+        @consul_power_args
+      end
```

Comment:
Isn't this exactly what Active Support's `class_attribute` does?

Reply:
I think `class_attribute` only copies values when you write to the attribute. In this case no one ever calls `consul_power_args=`, so the value would never be copied and changes in a child class would affect the parent class.